src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -327,12 +353,29 @@ bool CanIgnoreAllCollationElements(const UCollator* pColl, const UChar* lpStr, i
 
 }
 
-extern "C" ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName, SortHandle** ppSortHandle)
+void CreateSortHandle(SortHandle** ppSortHandle)
```

Comment:
>CreateSortHandle [](start = 5, length = 16)

what is the benefit making this a method by itself and not part of GlobalizationNative_GetSortHandle?

Reply:
> ResultCode GlobalizationNative_GetSortHandle(const char* lpLocaleName, SortHandle** ppSortHandle) 

is on line 373

Reply:
I don't have a strong feeling to change it but this initialization is not used anywhere else anyway and also when something fails we clean up in GlobalizationNative_GetSortHandle as we don't have a clean up method matching the allocation method.

Reply:
> we don't have a clean up method matching the allocation method

we have `GlobalizationNative_CloseSortHandle`, that does the cleanup

Reply:
What I mean, in CreateSortHandle we allocate the memory and then initialize the mutex there. and then in case of the failure, we free the mutex and memory in GlobalizationNative_GetSortHandle which is weird for me. my point is, either to have everything encapsulated inside GlobalizationNative_GetSortHandle or you create CreateSortHandle and DestroySortHandle kind. 

As I mentioned I am not strongly feeling about it so I'll leave it to you to decide if you need to change it or keep the current code.